firmware/thruna/BMS/Src/App/App_Accumulator.c

```diff
@@ -436,13 +437,30 @@ bool App_Accumulator_CheckFaults(struct Accumulator *const accumulator, struct T
     float max_allowable_cell_temp = MAX_CELL_DISCHARGE_TEMP_DEGC;
     float min_allowable_cell_temp = MIN_CELL_DISCHARGE_TEMP_DEGC;
 
+    float cell_voltage_minimum = 0;
+    float cell_voltage_sum;
+
+    for(uint8_t seg; seg <= 4; seg++){
+        float cell_voltage = App_Accumulator_GetAverageCellVoltage(accumulator, seg);
+        cell_voltage_sum += cell_voltage;
+        if(cell_voltage < cell_voltage_minimum){
+            cell_voltage_minimum = cell_voltage;
+        }
+    }
+    float cell_voltage_avg = cell_voltage_sum/4.0;
+    float segment_current = App_TractiveSystem_GetCurrent(ts);
+
+    float num_of_cells = (segment_current)*INTERNAL_CELL_RESISTANCE_OHMS/
+                        (cell_voltage_avg - cell_voltage_minimum + segment_current*INTERNAL_CELL_RESISTANCE_OHMS/3.0);
+
     // if we are charging, max cell temp is 45C not 60C
     if (App_TractiveSystem_GetCurrent(ts) > 3.0f)
     {
         max_allowable_cell_temp = MAX_CELL_CHARGE_TEMP_DEGC;
         min_allowable_cell_temp = MIN_CELL_CHARGE_TEMP_DEGC;
     }
 
+    bool blown_fuse_fault = num_of_cells < 2.5;
```

Comment:
make a macro for this 2.5 value at the top - also maybe drop a TODO about needing to do testing on the car to determine this 2.5 number. 
(unnecessary if we're going to determine that before we merge this in, but then you should write something about where the number comes from)